graph/flow/interval.go

```diff
@@ -0,0 +1,77 @@
+// Find the unique set of intervals of a control flow graph, as described in
+// Allen, Frances E., and John Cocke. "A program data flow analysis procedure."
+// Communications of the ACM 19.3 (1976): 137 [1].
+//
+// [1]: https://amturing.acm.org/p137-allen.pdf
+
+package flow
+
+import "gonum.org/v1/gonum/graph"
+
+// Intervals returns the unique set of intervals of the given control flow
+// graph. The returned set of intervals share the underlying nodes and edges of
+// the provided control flow graph.
+func Intervals(g Graph) []*Interval {
```

Comment:
Thinking about the API that you need to add above to allow a `graph.Directed` to be used here, ISTM that this should just be `func Intervals(root id, g graph.Directed) []*Interval`.

When `Dominators` is brought into flow, the signature should be updated to take an ID rather than a `graph.Node`.

Reply:
Yes, that makes sense. It would make the API less restrictive, as users would then not have to create a flow.Graph simply to use the function.